src/abt.sig

```diff
@@ -17,28 +41,65 @@ sig
   type metacontext = Metacontext.t
   type 'a spine = 'a Operator.Arity.Valence.Spine.t
 
+  (* The core type of the signature. This is the type of the ABTs that
+   * can be built from the given [operator]s, [variable]s, [symbol]s and
+   * [metavariable]s
+   *)
   type abt
+
+  (* An abs is an *abs*straction. It's the portion of syntax which binds
+   * variables and symbols for use in a term. It's conceptually separate
+   * from the abt type because it doesn't make sense to have a binder outside
+   * the context of some operator.
+   *)
   type abs
 
+  (* Equality on ABTs is the "right" sort of equality even though it's
+   * a little complicated. It considers terms up to
+   *  1. Alpha-varying bound variables
+   *  2. Apartness preserving renamings of bound symbols
+   *)
   structure Eq : EQ where type t = abt
 
   val freeVariables : abt -> (variable * sort) list
   val freeSymbols : abt -> (symbol * sort) list
+
+  (* This should be thought of as a "freeVariables" like function but
+   * in addition to just returning what metavariables are in scope,
+   * it also gives back information on the valence of each operator
+   * in a format that supports efficient random access
+   *)
   val metacontext : abt -> metacontext
 
   (* subst (N, x) M === [N/x]M *)
   val subst : abt * variable -> abt -> abt
 
-  (* metasubst (E, m) M === [E/m]M *)
+  (* metasubst (E, m) M === [E/m]M. Note that this does
+   * the appropriate normalization to specialize the given
+   * binder to the terms *)
   val metasubst : abs * metavariable -> abt -> abt
 
   (* rename (v, u) M === {v/u}M *)
   val rename : symbol * symbol -> abt -> abt
 
   (* Patterns for abstract binding trees. *)
+
+  (* A bview is a view of a binder. This is NOT an abt,
```

Comment:
- [ ] change to semicolon (`This is not an abt;`)
- [ ] let's say `abstraction` instead of `binding` or `binder`. (At some point we should perhaps rename `bview`, but I wasn't sure what sounded best.)